qiskit/algorithms/gradients/sampler_gradient_result.py

```diff
@@ -0,0 +1,37 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""
+Sampler result class
+"""
+
+from __future__ import annotations
+
+from typing import Any
+from dataclasses import dataclass
+
+from qiskit.result import QuasiDistribution
+
+
+@dataclass(frozen=True)
+class SamplerGradientResult:
+    """Result of SamplerGradient.
+
+    Args:
+        gradients: The gradients of the quasi distributions.
+        metadata: Additional information about the job.
+        run_options: run_options for the sampler. Currently, sampler's default run_options is not
+        included.
+    """
+
+    gradients: list[list[QuasiDistribution]]
```

Comment:
The gradient of probabilities or quasi-distribution is neither probability nor quasi-probability (sum of values is not always 1). So, It might be confusing to use `QuasiDistribution` to store the gradient. I saw there was a discussion on this topic in this PR, but I'm not sure the conclusion.

Reply:
yes here: https://github.com/Qiskit/qiskit-terra/pull/8528#discussion_r957352442 The conclusion was to not use the quasi distribution anymore 🙂 

@a-matsuo to answer your question in that conversation above: we have to use some sparse representation of the gradients, since we have 2^n elements in the distribution. So a dict would be good, or alternatively a `scipy.sparse` matrix. Maybe we could just choose what the current opflow gradients return?

Reply:
Oops, I forgot to change it. Thank you. let's use dict for now.